lib/testutil/testutil.go

```diff
@@ -59,3 +64,139 @@ type NoopCloser struct{}
 func (NoopCloser) Close() error {
 	return nil
 }
+
+func IfNotCI[T any](notCiValue T, ciValue T) T {
+	if os.Getenv("CI") == "true" {
+		return ciValue
+	} else {
+		return notCiValue
+	}
+}
+
+func ConcatSlices[T any](slices ...[]T) []T {
+	// TODO when go >= 1.22.0: Replace with slices.Concat
+	concatLen := 0
+	for _, s := range slices {
+		concatLen += len(s)
+	}
+	var concat []T = make([]T, concatLen)
+	i := 0
+	for _, s := range slices {
+		for _, el := range s {
+			concat[i] = el
+			i += 1
+		}
+	}
+	return concat
+}
+
+func IfExpr[T any](expr bool, then T, els T) T {
+	if expr {
+		return then
+	} else {
+		return els
+	}
+}
+
+func AssertTrue(t *testing.T, testFailFunc func(string, ...any), a bool, sprintfArgs ...any) {
```

Comment:
DeepSource was complaining about very high cyclomatic complexity due to all the `if !... { t.Error(...) }`s you end up with with vanilla "assertions", so this was my way of working around that. I didn't want to impose another dependency without approval (and I don't know the Go ecosystem well enough to know which are any good), but I'm happy to do so if you like.

Shall I use https://github.com/stretchr/testify for this, then?

Reply:
No, use vanilla expressions like we do in all other tests, please.

Reply:
By "vanilla" do you mean just the raw `if !(...) { t.Errorf/Fatalf(...) }` (i.e., inlining these `Assert*` and `FatalIfErr` functions at each call site)? If so, should I not worry about the DeepSource lints on cyclomatic complexity?

Reply:
IMHO, cyclomatic complexity is specifically about a function being testable.  This is a test function, it will not be tested itself.  There is only one desired outcome which must work - the all success case.  Coverage of all other possible branches is moot, since whatever check triggers, will show up as a failure and be investigated individually.

So yeah, just ignore those lints in this case.

Reply:
Ok. Fixed in commits f56169b21 to 7e6e0999b.